### ISSUE TITLE
Feature request: drop in syscall replacement

### ISSUE THREAD

**Issue by lloesche** [zerotier/libzt#89]
It'd be neat if I could LD_PRELOAD a shim that reliably replaces relevant syscalls with libzt counterparts. That would allow me to tunnel traffic of unmodified binaries purely in user space without the need for any elevated privileges. Esp. useful for restricted containers.

**Comment by joseph-henry** [comment]
Agreed. That used to be included in the old builds. I've had a few people request this recently, I might whip something up. I'll update this ticket when I do.
